src/middlewared/middlewared/plugins/jail.py

```diff
@@ -827,8 +844,46 @@ def do_update(self, jail, options):
         if name:
             iocage.rename(name)
 
+        new = self.middleware.call_sync('jail._get_instance', name or jail['id'])
+        if (
+            any(new[k] != jail[k] for k in ('vnet', 'nat')) or (
+                (new['vnet'] and new['vnet_default_interface'] != jail['vnet_default_interface']) or (
+                    new['nat'] and new['nat_interface'] != jail['nat_interface']
+                )
+            )
+        ) and not self.middleware.call_sync('system.is_freenas') and self.middleware.call_sync('failover.licensed'):
+            # We do this to to re-enable capabilities
+            self.enable_capabilities_for_nic(jail)
         return True
 
+    @private
+    def enable_capabilities_for_nic(self, jail_config):
+        if not self.middleware.call_sync('system.is_freenas') and self.middleware.call_sync(
+            'failover.licensed'
+        ) and self.middleware.call_sync('failover.config')['disabled']:
```

Comment:
Should we raise in error if HA is not disabled?

Reply:
I was confused initially, thinking more about it, i think it would make more sense if we did raise an error here.